### ISSUE TITLE
Help `@ParameterizedTest` users with better error message

### ISSUE THREAD

**Issue by iaincd** [diffplug/selfie#140]
At present, a `@ParameterizedTest` with arguments that in any way affect the appearance of a snapshot fail on the latter iteration(s), due to Selfie wanting the initial snapshot to match all iterations.

It would be neat if this could be detected, and for the runner to then append the additional unique state, instead — either labelled numerically within the snapshot, or maybe numbered + the same argument naming convention as JUnit, i.e. respecting `Arguments.of(Named.of(...))`, or else falling back to `toString()`. 

Disclaimer: I do not know whether JUnit 5 even provides the hooks necessary to see which iteration/argument set is currently executing, but I would hope this is possible.

I'm also totally open to the possibility that you might view this as a test design smell, but I for one get a lot of mileage out of `@ParameterizedTest`s where the snapshot-candidate is not static between iterations.

**Comment by nedtwigg** [comment]
I think `@ParameterizedTest` is great, and I think JUnit 5 gives us the hooks we need to do what you want, but I don't think we need a new selfie feature.

Whenever you do `.toMatchDisk()` you can optionally supply a parameter, [`.toMatchDisk(String sub)`](https://kdoc.selfie.dev/selfie-lib/com.diffplug.selfie/-selfie/-disk-selfie/to-match-disk). So the workaround you can do right now is:

```java
@ParameterizedTest @ValueSource(ints = {1, 2, 5})
void testDataStructureOfLength(int size) {
  var underTest = initialize()
  ... complicated stuff ...
  expectSelfie(underTest).toMatchDisk("size = " + size)
}
```

The reason I like this is that it gives you control over how your snapshots show up in your file. It can be any of:

```
╔═ testDataStructureOfLength/1 ═╗
╔═ testDataStructureOfLength/size=1 ═╗
╔═ testDataStructureOfLength/size = 1 ═╗
╔═ testDataStructureOfLength/size = 01 ═╗
```

One of the things we're trying to do with selfie is make everything explicit. Like spotless, you just give selfie strings, no magic implicit transformations.

Another justification for this is inline snapshots. There's no way selfie can automatically fork the source code for a parameterized test, but you can fork it yourself!

```java
@ParameterizedTest @ValueSource(ints = {1, 2, 5})
void testDataStructureOfLength(int size) {
  var underTest = initialize()
  ... complicated stuff ...
  if (size == 1) {
    expectSelfie(underTest).toBe("[1]")
  } else if size == 5) {
    expectSelfie(underTest).toBe("[1, 1, 2, 3, 5]")
  }
  expectSelfie(underTest).toMatchDisk("size = " + size)
}
```

We could make the error message a lot better though.

https://github.com/diffplug/selfie/blob/b151801d29b54559949d11797931a9e241e6b1e3/selfie-lib/src/commonMain/kotlin/com/diffplug/selfie/guts/WriteTracker.kt#L62

What do you think of these messages?

```
Snapshot was set to multiple values!\n  
  first time: ${existing.callStack.location.ideLink(layout)}\n 
   this time: ${call.location.ideLink(layout)}
(for disk) You can fix this with `.toMatchDisk(String sub)` and passing a unique value for sub.
(for inline) You can fix this by doing an `if` before the assertion to separate the cases.
  if (blah) {
    expectSelfie(underTest).toBe("shoopy doopy")
  } else {
    expectSelfie(underTest).toBe("gloopy gloppy")
  }
```

**Comment by iaincd** [comment]
Aha, excellent! I think the facility you describe sounds just fine, and adding the relevant hint to the error messages would do nicely.

**Comment by nedtwigg** [comment]
Better error message shipped in `1.1.2`.
